File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    maxDepth(root) {
        
        if(!root) return 0;
        
        let road = new Map();
        let q = [root];
        let control = 0;

        while(control < q.length){
            let currNode = q[control];
            if(currNode.left){
                q.push(currNode.left);
                road.set(currNode.left, currNode)
            }
               
            if(currNode.right){
                q.push(currNode.right);
                road.set(currNode.right, currNode)
            }
            
            control++;
        }
        
        let lastVisited = q.pop();
        let res = 1;
        while(lastVisited!=root){
            lastVisited=road.get(lastVisited);
            res++;
        }

        
        return res;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func levelOrder(root *TreeNode) [][]int {

    var res [][]int

    if root == nil {
        return [][]int{}
    }

    q := []*TreeNode{}
    q = append(q, root)

    
     for len(q) > 0 {
        var level []int

        for _, el := range(q) {

            node := q[0]
            q = q[1:len(q)]

            if node != nil {
                level = append(level, node.Val)
                q = append(q, el.Left)
                q = append(q, el.Right)
            }   
        }

        if len(level) > 0 {
            res = append(res, level)
        }

     }

     return res
}
